### PR TITLE
chore: migrate template to version 2

### PR DESCRIPTION
**Pull request by tgpetrov**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,8 +5,8 @@ npm-debug.log
 vendor*.ts
 *.js
 !tools/*.js
+!hooks/**/*.js
```
Comment (sis0k0) [inline .gitignore]: What's the reason for keeping the `hooks/.js` files in git?

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
+npm-debug.log
+.DS_Store
+
+*.js.map
+vendor*.ts
```
Comment (sis0k0) [inline .npmignore]: The vendor files are no longer added by the webpack plugin.

File: hooks/after-createProject/after-createProject.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+const fs = require("fs");
```
Comment (sis0k0) [inline hooks/after-createProject/after-createProject.js]: Do we still need this hook with the new template structure?

File: .gitignore (hunk 1 above)
Comment (tgpetrov) [reply .gitignore]: We still need scripts to apply project modifications, they used to be in tools, but now with template v2, they [should be in hooks](https://gist.github.com/rosen-vladimirov/db05391603646ae3b40e13563cc8c4f2#project-creation-v2).

File: .npmignore (hunk 2 above)
Comment (tgpetrov) [reply .npmignore]: We can do a clean up of some files that don't need to be ignored any longer (I'm not sure about 'tmp/' and 'typings/'), this content is just an extended copy of the existing .gitignore file in order to exclude the template assets from the package.

File: hooks/after-createProject/after-createProject.js (hunk 3 above)
Comment (tgpetrov) [reply hooks/after-createProject/after-createProject.js]: We use it in order to have one .gitignore file for the template itself and another that gets included in the app created from the template. For example to include hooks in the template and exclude then the produced app. Also to remove the tools directory.

**Pull request merged**